### PR TITLE
added dialect

### PR DESCRIPTION
**Pull request by DvirDukhan**

### CODE REVIEW

**Review by chayim — DISMISSED** [review]

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 1]
```diff
@@ -454,7 +454,8 @@ public void testHNSWVVectorSimilarity() {
     Query query =  new Query("*=>[KNN 2 @v $vec]")
         .addParam("vec", "aaaaaaaa")
         .setSortBy("__v_score", true)
-        .returnFields("__v_score");
+        .returnFields("__v_score")
+        .dialect(2);
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:458]: If this is not necessary to pass the test, please remove this change.

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 2]
```diff
@@ -477,12 +478,142 @@ public void testFlatVectorSimilarity() {
...
     Document doc1 = client.ftSearch(index, query).getDocuments().get(0);
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:483]: If this is not necessary to pass the test, please remove this change.

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 3]
```diff
@@ -492,7 +623,7 @@ public void testQueryParams() {
     client.hset("2", "numval", "2");
     client.hset("3", "numval", "3");
 
-    Query query =  new Query("@numval:[$min $max]").addParam("min", 1).addParam("max", 2);
+    Query query =  new Query("@numval:[$min $max]").addParam("min", 1).addParam("max", 2).dialect(2);
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:626]: If this is not necessary to pass the test, please remove this change.
If you want to have this test as well, just add a new one.

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 4]
```diff
@@ -477,12 +478,142 @@ public void testFlatVectorSimilarity() {
...
+    try {
+      assertEquals("OK", client.ftConfigSet("DEFAULT_DIALECT", "0"));
+      fail();
+    }
+    catch (JedisDataException ex) {
+
+    }
+
+    try {
+      assertEquals("OK", client.ftConfigSet("DEFAULT_DIALECT", "3"));
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:510]: Remove assert.

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 5]
```diff
@@ -477,12 +478,142 @@ public void testFlatVectorSimilarity() {
...
+      }}
+      , client.ftConfigGet("DEFAULT_DIALECT"));
+    assertEquals("OK", client.ftConfigSet("DEFAULT_DIALECT", "2"));
+    assertEquals(new HashMap<String, String>() {{
+      put("DEFAULT_DIALECT", "2");
+      }}
+      , client.ftConfigGet("DEFAULT_DIALECT"));
+    
+    try {
+      assertEquals("OK", client.ftConfigSet("DEFAULT_DIALECT", "0"));
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:502]: Remove assert.

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 6]
```diff
@@ -477,12 +478,142 @@ public void testFlatVectorSimilarity() {
...
+        .dialect(2);
     Document doc1 = client.ftSearch(index, query).getDocuments().get(0);
     assertEquals("a", doc1.getId());
     assertEquals("0", doc1.get("__v_score"));
   }
 
+  @Test
+  public void testDialectConfig() throws Exception{
+    assertEquals("OK", client.ftConfigSet("DEFAULT_DIALECT", "1"));
+    assertEquals(new HashMap<String, String>() {{
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:491]: Use Collections.singletonMap(...)

File: src/test/java/redis/clients/jedis/modules/search/SearchTest.java [hunk 7]
```diff
@@ -477,12 +478,142 @@ public void testFlatVectorSimilarity() {
...
+      put("DEFAULT_DIALECT", "1");
+      }}
+      , client.ftConfigGet("DEFAULT_DIALECT"));
+    assertEquals("OK", client.ftConfigSet("DEFAULT_DIALECT", "2"));
+    assertEquals(new HashMap<String, String>() {{
```
Comment (sazzad16) [inline src/test/java/redis/clients/jedis/modules/search/SearchTest.java:496]: Use Collections.singletonMap(...)

**Review by sazzad16 — CHANGES_REQUESTED** [review]

**Review by sazzad16 — APPROVED** [review]

**Pull request merged**